### PR TITLE
Doc improvements

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

File: doc/source/API/Constants.rst [hunk 1]
```diff
@@ -2,12 +2,6 @@ Constants
 ==========
 This section lists all the useful constants used in pyaedt.
```
Comment (PipKat) [inline doc/source/API/Constants.rst:3]:
```suggestion
This section lists commonly used constants in PyAEDT.
```

File: doc/source/API/Setup.rst [hunk 2]
```diff
@@ -2,17 +2,28 @@ Setup
...
-   SetupTemplates.SweepHFSS3DLayout
-   SetupTemplates.SweepQ3D
-   SetupTemplates.SetupKeys
+   Setup
+   Setup3DLayout
+   SetupCircuit
+
+Sweep Classes
+=============
+This section lists the Sweeos classes and their default values.
```
Comment (PipKat) [inline doc/source/API/Setup.rst:18]:
```suggestion
This section lists Sweep classes and their default values.
```

File: examples/00-basic-examples/01_edb_example.py [hunk 3]
```diff
@@ -152,7 +152,7 @@
 
 ###############################################################################
 # Create a Voltage Source and Siwave DC IR Simulation
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 # This example creates a Voltage Source and then setup a DCIR Analysis.
```
Comment (PipKat) [inline examples/00-basic-examples/01_edb_example.py:156]:
```suggestion
# This example creates a voltage source and then sets up a DCIR analysis.
```

File: pyaedt/modeler/Model3D.py [hunk 4]
```diff
@@ -14,9 +14,11 @@ class Modeler3D(GeometryModeler):
     ----------
     application : :class:`pyaedt.application.Analysis3D.FieldAnalysis3D`
```
Comment (PipKat) [inline pyaedt/modeler/Model3D.py:16]:
```suggestion
```

File: pyaedt/modeler/Primitives.py [hunk 5]
```diff
@@ -107,14 +107,12 @@ class Polyline(Object3d):
 
     Parameters
     ----------
-    parent : :class:`pyaedt.modeler.Primitives.Primitives`
+    primitives : :class:`pyaedt.modeler.Primitives.Primitives`
         Pointer to the parent Primitives object.
     src_object : optional
         The default is ``None``.
     position_list : list, optional
         List of positions in the ``[x, y, z]`` form. The default is ``None``.
```
Comment (PipKat) [inline pyaedt/modeler/Primitives.py:115]:
```suggestion
        List of positions in the ``[x, y, z]`` format. The default is ``None``.
```

File: pyaedt/modeler/Primitives2D.py [hunk 6]
```diff
@@ -7,6 +7,15 @@ class Primitives2D(Primitives, object):
 
     This class is inherited in the caller application and is accessible through the primitives variable part
     of modeler object( eg. ``hfss.modeler.primitives`` or ``icepak.modeler.primitives``).
```
Comment (PipKat) [inline pyaedt/modeler/Primitives2D.py:9]:
```suggestion
    of modeler object (for example, ``hfss.modeler.primitives`` or ``icepak.modeler.primitives``).
```

Comment (PipKat) [reply pyaedt/modeler/Primitives2D.py:9]: Ansys doc styles say not to use Latin expressions.

File: pyaedt/modeler/Primitives2D.py [hunk 7]
```diff
@@ -7,6 +7,15 @@ class Primitives2D(Primitives, object):
...
+
+
+    Examples
+    --------
+    Basic usage demonstrated with an Q2d or Maxwell2d design:
```
Comment (PipKat) [inline pyaedt/modeler/Primitives2D.py:14]:
```suggestion
    Basic usage demonstrated with a Q2D or Maxwell 2D design:
```

File: pyaedt/modeler/Primitives3D.py [hunk 8]
```diff
@@ -19,6 +19,13 @@ class Primitives3D(Primitives, object):
     modeler : str
         Name of the modeler.
 
+    Examples
+    --------
+    Basic usage demonstrated with an HFSS, Maxwell3d, Icepak, Q3d or Mechanical design:
```
Comment (PipKat) [inline pyaedt/modeler/Primitives3D.py:24]:
```suggestion
    Basic usage demonstrated with an HFSS, Maxwell 3D, Icepak, Q3D, or Mechanical design:
```

File: pyaedt/modeler/Primitives3DLayout.py [hunk 9]
```diff
@@ -27,6 +27,13 @@ class Primitives3DLayout(object):
     modeler : :class:`pyaedt.modeler.Model3DLayout.Modeler3DLayout`
         Name of the modeler.
 
+    Examples
+    --------
+    Basic usage demonstrated with an HFSS 3d Layout design:
```
Comment (PipKat) [inline pyaedt/modeler/Primitives3DLayout.py:32]:
```suggestion
    Basic usage demonstrated with an HFSS 3D Layout design:
```

File: pyaedt/modeler/PrimitivesCircuit.py [hunk 10]
```diff
@@ -10,6 +10,14 @@ class CircuitComponents(object):
     """CircutComponents class.
 
     This is the common class for managing all circuit components for Nexxim and Simplorer.
+
+    Examples
+    --------
+    Basic usage demonstrated with an Circuit design:
```
Comment (PipKat) [inline pyaedt/modeler/PrimitivesCircuit.py:16]:
```suggestion
    Basic usage demonstrated with a Circuit design:
```

File: pyaedt/modeler/PrimitivesNexxim.py [hunk 11]
```diff
@@ -14,6 +14,13 @@ class NexximComponents(CircuitComponents):
         Inherited parent object.
     modeler :
 
+    Examples
+    --------
+    Basic usage demonstrated with an Circuit design:
```
Comment (PipKat) [inline pyaedt/modeler/PrimitivesNexxim.py:19]:
```suggestion
    Basic usage demonstrated with a Circuit design:
```

File: pyaedt/modules/AdvancedPostProcessing.py [hunk 12]
```diff
@@ -77,6 +77,13 @@ class PostProcessor(Post):
     app :
         Inherited parent object.
 
+    Examples
+    --------
+    Basic usage demonstrated with an HFSS, Maxwell or any other design:
```
Comment (PipKat) [inline pyaedt/modules/AdvancedPostProcessing.py:82]:
```suggestion
    Basic usage demonstrated with an HFSS, Maxwell, or any other design:
```

Comment (PipKat) [reply pyaedt/modules/AdvancedPostProcessing.py:82]: Ansys doc style is to use the Oxford comma, which is a comma before the "and" or the "or" in a series of items.

File: pyaedt/modules/PostProcessor.py [hunk 13]
```diff
@@ -480,6 +480,67 @@ def data_imag(self, expression=None, convert_to_SI=False):
             )
         return sol
 
+    @aedt_exception_handler
+    def is_real_only(self, expression=None):
+        """Check if the expression has only real values or not.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py]:
```suggestion
        """Check whether the expression has only real values.
```

File: pyaedt/modules/PostProcessor.py [hunk 14]
```diff
@@ -480,6 +480,67 @@ def data_imag(self, expression=None, convert_to_SI=False):
...
+        return True
+
+    @aedt_exception_handler
+    def export_data_to_csv(self, output, delimiter=";"):
+        """Save to output csv file the Solution Data.
+
+        Parameters
+        ----------
+        output : str,
+            Full path to csv file.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py]:
```suggestion
            Full path to the CSV file.
```

File: pyaedt/modules/PostProcessor.py [hunk 15]
```diff
@@ -480,6 +480,67 @@ def data_imag(self, expression=None, convert_to_SI=False):
...
+        if not expression:
+            expression = self.expressions[0]
+        for e, v in self.solutions_data_imag[expression].items():
+            if float(v) != 0.0:
+                return False
+        return True
+
+    @aedt_exception_handler
+    def export_data_to_csv(self, output, delimiter=";"):
+        """Save to output csv file the Solution Data.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py]:
```suggestion
        """Save the solution data to CSV file.
```

File: pyaedt/modules/PostProcessor.py [hunk 16]
```diff
@@ -480,6 +480,67 @@ def data_imag(self, expression=None, convert_to_SI=False):
...
+        Parameters
+        ----------
+        expression : str, optional
+            Name of the expression. The default is ``None``,
+            in which case the first expression is used.
+
+        Returns
+        -------
+        bool
+            ``True`` if the Solution Data for specific expression contains only real values.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py]:
```suggestion
            ``True`` if the solution data for a specific expression contains only real values.
```

File: pyaedt/modules/PostProcessor.py (hunk 14 above)
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py]:
```suggestion
            Full path to the CSV file.
```

File: pyaedt/modules/PostProcessor.py [hunk 17]
```diff
@@ -480,6 +480,67 @@ def data_imag(self, expression=None, convert_to_SI=False):
...
+    @aedt_exception_handler
+    def export_data_to_csv(self, output, delimiter=";"):
+        """Save to output csv file the Solution Data.
+
+        Parameters
+        ----------
+        output : str,
+            Full path to csv file.
+        delimiter : str,
+            CSV Delimiter. Default is ``";"``.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py]:
```suggestion
            CSV delimiter. Default is ``";"``.
```

File: pyaedt/modules/PostProcessor.py [hunk 18]
```diff
@@ -768,10 +829,10 @@ def change_plot_scale(self, minimum_value, maximum_value, is_log=False, is_db=Fa
 
     @aedt_exception_handler
     def export_image(self, full_path=None, width=1920, height=1080, orientation="isometric", display_wireframe=True):
-        """Save an image of active Plot.
+        """Export the active plot to an image file.
 
         .. note::
-           There are some limitation on HFSS3DLayout.
+            There are some limitation on HFSS3DLayout.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:835]:
```suggestion
            There are some limitations on HFSS 3D Layout.
```

File: pyaedt/modules/PostProcessor.py [hunk 19]
```diff
@@ -768,10 +829,10 @@ def change_plot_scale(self, minimum_value, maximum_value, is_log=False, is_db=Fa
...
 
         full_path : str, optional
             Path where image will be saved. It supports png and gif format.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:838]:
```suggestion
            Path for saving the image file. PNG and GIF formats are supported.
```

File: pyaedt/modules/PostProcessor.py [hunk 20]
```diff
@@ -806,14 +868,15 @@ def export_image_from_aedtplt(self, export_path=None, view="isometric", plot_mes
         """Save an image of Active Plot using PyVista.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:868]:
```suggestion
        """Save an image of the active plot using PyVista.
```

File: pyaedt/modules/PostProcessor.py [hunk 21]
```diff
@@ -780,7 +841,8 @@ def export_image(self, full_path=None, width=1920, height=1080, orientation="iso
         height : int, optional
             Plot Height.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:842]:
```suggestion
            Plot height.
```

File: pyaedt/modules/PostProcessor.py [hunk 22]
```diff
@@ -806,14 +868,15 @@ def export_image_from_aedtplt(self, export_path=None, view="isometric", plot_mes
         """Save an image of Active Plot using PyVista.
 
         .. note::
-           Only working in CPython with PyVista Module Installed.
+            Only working in CPython with PyVista Module Installed.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:871]:
```suggestion
            This method is only working if the CPython with PyVista module is installed.
```

File: pyaedt/modules/PostProcessor.py [hunk 23]
```diff
@@ -806,14 +868,15 @@ def export_image_from_aedtplt(self, export_path=None, view="isometric", plot_mes
...
 
         Parameters
         ----------
         export_path : str, optional
             Path where image will be saved
         view : str, optional
-            View of the exported plot. Options are ``isometric``, ``top``, ``front``, ``left``, ``all``.
+            View of the exported plot. Options are ``isometric``,
+            ``top``, ``front``, ``left``, ``all``.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:879]:
```suggestion
            ``top``, ``front``, ``left``, and ``all``.
```

File: pyaedt/modules/PostProcessor.py [hunk 24]
```diff
@@ -1345,6 +1408,13 @@ class PostProcessor(PostProcessorCommon, object):
         Inherited parent object. The parent object must provide the members
         `_modeler`, `_desktop`, `_odesign`, and `logger`.
 
+    Examples
+    --------
+    Basic usage demonstrated with an HFSS, Maxwell or any other design:
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:1413]:
```suggestion
    Basic usage demonstrated with an HFSS, Maxwell, or any other design:
```

File: pyaedt/modeler/Primitives.py [hunk 25]
```diff
@@ -107,14 +107,12 @@ class Polyline(Object3d):
 
     Parameters
     ----------
-    parent : :class:`pyaedt.modeler.Primitives.Primitives`
+    primitives : :class:`pyaedt.modeler.Primitives.Primitives`
         Pointer to the parent Primitives object.
```
Comment (PipKat) [inline pyaedt/modeler/Primitives.py:111]: Is "Primitives" necessary in the description? It is not included in line 812 below.

File: pyaedt/modeler/Primitives.py [hunk 26]
```diff
@@ -107,14 +107,12 @@ class Polyline(Object3d):
...
     src_object : optional
         The default is ``None``.
     position_list : list, optional
         List of positions in the ``[x, y, z]`` form. The default is ``None``.
-    object_id : optional
-        The default is ``None``.
     segment_type : optional
         The default is ``None``.
```
Comment (PipKat) [inline pyaedt/modeler/Primitives.py:117]: Can a description be added here?

File: pyaedt/modeler/PrimitivesCircuit.py [hunk 27]
```diff
@@ -10,6 +10,14 @@ class CircuitComponents(object):
     """CircutComponents class.
 
     This is the common class for managing all circuit components for Nexxim and Simplorer.
```
Comment (PipKat) [inline pyaedt/modeler/PrimitivesCircuit.py:12]:
```suggestion
    Manages all circuit components for Nexxim and Simplorer.
```

File: pyaedt/modeler/PrimitivesNexxim.py [hunk 28]
```diff
@@ -14,6 +14,13 @@ class NexximComponents(CircuitComponents):
         Inherited parent object.
     modeler :
```
Comment (PipKat) [inline pyaedt/modeler/PrimitivesNexxim.py:15]: Can a description be added for modeler?

File: pyaedt/modules/PostProcessor.py [hunk 29]
```diff
@@ -780,7 +841,8 @@ def export_image(self, full_path=None, width=1920, height=1080, orientation="iso
         height : int, optional
             Plot Height.
         orientation : str, optional
-            View of the exported plot. Options are ``isometric``, ``top``, ``bottom``, ``right``, ``left``, ``front``,
+            View of the exported plot. Options are ``isometric``,
+            ``top``, ``bottom``, ``right``, ``left``, ``front``,
             ``back``any custom orientation.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:846]:
```suggestion
            ``back``, and any custom orientation.
```

File: pyaedt/modules/PostProcessor.py [hunk 30]
```diff
@@ -806,14 +868,15 @@ def export_image_from_aedtplt(self, export_path=None, view="isometric", plot_mes
...
         Parameters
         ----------
         export_path : str, optional
             Path where image will be saved
         view : str, optional
-            View of the exported plot. Options are ``isometric``, ``top``, ``front``, ``left``, ``all``.
+            View of the exported plot. Options are ``isometric``,
+            ``top``, ``front``, ``left``, ``all``.
         plot_mesh : bool, optional
             Plot Mesh.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:881]:
```suggestion
            Plot mesh.
```

File: pyaedt/modules/PostProcessor.py [hunk 31]
```diff
@@ -1345,6 +1408,13 @@ class PostProcessor(PostProcessorCommon, object):
         Inherited parent object. The parent object must provide the members
         `_modeler`, `_desktop`, `_odesign`, and `logger`.
```
Comment (PipKat) [inline pyaedt/modules/PostProcessor.py:1409]:
```suggestion
        ``_modeler``, ``_desktop``, ``_odesign``, and ``logger``.
```

File: pyaedt/modeler/PrimitivesCircuit.py (hunk 27 above)
Comment (PipKat) [reply pyaedt/modeler/PrimitivesCircuit.py:12]: I changed to fit the format for descriptions in the summary table listing all classes.

**Review by PipKat — COMMENTED** [review]
I reviewed the doc strings.

File: pyaedt/modeler/Primitives.py (hunk 25 above)
Comment (maxcapodi78) [reply pyaedt/modeler/Primitives.py:111]: yes it is. Thanks

File: pyaedt/modeler/Primitives.py (hunk 26 above)
Comment (maxcapodi78) [reply pyaedt/modeler/Primitives.py:117]: Done

File: pyaedt/modeler/PrimitivesNexxim.py (hunk 28 above)
Comment (maxcapodi78) [reply pyaedt/modeler/PrimitivesNexxim.py:15]: done

File: doc/source/API/Application.rst [hunk 32]
```diff
@@ -1,6 +1,36 @@
-Application
-===========
+AEDT Applications
+=================
 The PyAEDT API includes these classes for applications and modules.
+The application has to be initialized by the user. All other classes and methods are inherited into application class.
+Desktop application is implicitily launched in any of the other applications.
```
Comment (MaxJPRey) [inline doc/source/API/Application.rst:5]:
```suggestion
Desktop application is implicitly launched in any of the other applications.
```

File: doc/source/API/Constants.rst [hunk 33]
```diff
@@ -1,8 +1,21 @@
...
+Example of constants usage:
+
+.. code:: python
+
+    from pyaedt import constants
+    ipk = Icepak()
+    # Use of AXIS Constant
+    cyl = ipk.modeler.primitives.create_cylinder(constants.AXIS.X, [0,0,0],10,3)
+    # Use of PLANE Constant
+    ipk.modeler.split(cyl, constants.PLANE.YZ, sides="Both")
```
Comment (MaxJPRey) [inline doc/source/API/Constants.rst:15]:
```suggestion
    cylinder = ipk.modeler.primitives.create_cylinder(constants.AXIS.X, [0,0,0],10,3)
    # Use of PLANE Constant
    ipk.modeler.split(cylinder, constants.PLANE.YZ, sides="Both")
```

File: doc/source/API/Primitives.rst [hunk 34]
```diff
@@ -54,15 +114,80 @@ The ``Primitives`` module includes these classes:
 
 Objects
 ~~~~~~~
+Those class define the objects properties for 3D and 2D Solvers (excluding ``Hfss3dLayout``).
```
Comment (MaxJPRey) [inline doc/source/API/Primitives.rst:117]:
```suggestion
Those classes define the objects properties for 3D and 2D Solvers (excluding ``Hfss3dLayout``).
```

File: doc/source/API/Primitives.rst [hunk 35]
```diff
@@ -54,15 +114,80 @@ The ``Primitives`` module includes these classes:
...
+
+
+.. code:: python
+
+    from pyaedt import Hfss
+    app = Hfss(specified_version="2021.1",
+                 non_graphical=False, new_desktop_session=True,
+                 close_on_exit=True, student_version=False)
+
+    # this call return the CoordinateSystem Object lists
```
Comment (MaxJPRey) [inline doc/source/API/Primitives.rst:176]:
```suggestion
    # this call returns the CoordinateSystem Object lists
```

File: doc/source/API/Primitives.rst [hunk 36]
```diff
@@ -54,15 +114,80 @@ The ``Primitives`` module includes these classes:
...
+    cs = app.modeler.coordinate_systems
+
+    # this call return a Primitives3D Object
```
Comment (MaxJPRey) [inline doc/source/API/Primitives.rst:179]:
```suggestion
    # this calls return a Primitives3D Object
```

File: doc/source/API/Setup.rst [hunk 37]
```diff
@@ -1,18 +1,53 @@
...
+Setup object is accessible through the ``create_setup`` method and ``setups`` object list.
+.. code:: python
 
-.. currentmodule:: pyaedt.modules
+    from pyaedt import Hfss
+    app = Hfss(specified_version="2021.1",
+                 non_graphical=False, new_desktop_session=True,
+                 close_on_exit=True, student_version=False)
+
+    # this call return the Setup Class
```
Comment (MaxJPRey) [inline doc/source/API/Setup.rst:15]:
```suggestion
    # this call returns the Setup Class
```

File: doc/source/API/Setup.rst [hunk 38]
```diff
@@ -1,18 +1,53 @@
...
+    my_setup = app.setups[0]
+
+
+    # this call return a Setup Object
```
Comment (MaxJPRey) [inline doc/source/API/Setup.rst:19]:
```suggestion
    # this call returns a Setup Object
```

**Review by MaxJPRey — APPROVED** [review]

**Pull request merged**